datasets/wiki_lingua/wiki_lingua.py

```diff
@@ -0,0 +1,209 @@
+        datasets.BuilderConfig(name="turkish", version=VERSION, description="A subset of article-summary in Turkish"),
+        datasets.BuilderConfig(
+            name="vietnamese", version=VERSION, description="A subset of article-summary in Vietnamese"
+        ),
+    ]
+
+    DEFAULT_CONFIG_NAME = "english"
+
+    def _info(self):
+        if self.config.name == "english":
+            features = datasets.Features(
+                {
+                    "url": datasets.Value("string"),
+                    "article": datasets.Sequence(
+                        {
+                            "section_name": datasets.Value("string"),
+                            "document": datasets.Value("string"),
+                            "summary": datasets.Value("string"),
+                        }
+                    ),
+                }
+            )
+        else:
+            features = datasets.Features(
+                {
+                    "url": datasets.Value("string"),
+                    "article": datasets.Sequence(
+                        {
+                            "section_name": datasets.Value("string"),
+                            "document": datasets.Value("string"),
+                            "summary": datasets.Value("string"),
+                            "english_url": datasets.Value("string"),
+                            "english_section_name": datasets.Value("string"),
+                        }
+                    ),
+                }
+            )
+
+        return datasets.DatasetInfo(
+            # This is the description that will appear on the datasets page.
+            description=_DESCRIPTION,
+            # This defines the different columns of the dataset and their types
+            features=features,  # Here we define them above because they are different between the two configurations
+            # If there's a common (input, target) tuple from the features,
+            # specify them here. They'll be used if as_supervised=True in
+            # builder.as_dataset.
+            supervised_keys=None,
+            # Homepage of the dataset for documentation
+            homepage=_HOMEPAGE,
+            # License for the dataset if available
+            license=_LICENSE,
+            # Citation for the dataset
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager):
+        """Returns SplitGenerators."""
+        my_urls = _URLs[self.config.name]
+        # The domain specific files are all in pkl format so when
+        # the download manager downloads it, the path returned is
+        # the actual file path where in tests we get the dir
+        data_dir = dl_manager.download_and_extract(my_urls)
+        # See create_dummy.py to create new dummy data
+        if os.path.isdir(data_dir):
+            files = glob(f"{data_dir}/*")
+            # Will pickup the 1st file. Not sure if this is fine though
+            train_fname = files[0]
```

Comment:
do you really need that ?

If I understand correctly the downloaded files are just .pkl files so you don't need to use glob to look for files inside a directory ?

Reply:
It is only used when running the tests. 

I am not sure why the tests return the dir instead of full file. Hence, I added the check if its a dir then use first file inside the directory. For example, when I run `RUN_SLOW=1 pytest tests/test_dataset_common.py::LocalDatasetTest::test_load_dataset_all_configs_wiki_lingua`
I get the directory in _split_generators `/var/folders/th/0p_qpf7j36q43jhvs5y2r2c80000gn/T/tmpq3_3jsji/extracted/7ba2849326690fd0d15e8d1e4f23ac2b2dc3ddbc4499d723d84da360d0048926/dummy_data`.

It seems unique to this dataset. I raised another PR where I am adding jsonl files and it works fine there.

Reply:
It is returning the parent directory because it couldn't find the expected file in the dummy_data.zip

Indeed it looks like the names of the dummy pkl files are wrong.
For example for the arabic config, the dummy data file is named
```
uc?export%3download%26id%31__EjA6oZsgXQpggPm-h54jZu3kP6Y6zu
```
instead of 
```
uc%3Fexport%3Ddownload%26id%3D1__EjA6oZsgXQpggPm-h54jZu3kP6Y6zu
```


You can get the right names by running
```
datasets-cli dummy_data ./datasets/wiki_lingua
```

Can you please rename the dummy data files in each config ?

Once the names are fixed you should be able to rename the `glob` trick and it should work :)

Reply:
Indeed 😅, the filenames in dummy data were incorrect. Thanks for the pointer @lhoestq . Tests work fine without the glob. I have added the clean version in my last commit😃